Lab 1

In [3]:
import pandas as pd
names = ['Triston Hudgins','Shijo Joseph','Douglas Yip']
teaminfo = pd.DataFrame()
teaminfo['Team'] = names
teaminfo

Team
0  Triston Hudgins
1     Shijo Joseph
2      Douglas Yip

• Business Understanding (10 points total).
• Describe the purpose of the data set you selected (i.e., why was this data collected in
the first place?). Describe how you would define and measure the outcomes from the
dataset. That is, why is this data important and how do you know if you have mined
useful knowledge from the dataset? How would you measure the effectiveness of a
good prediction algorithm? Be specific

In [12]:
# load the Lehman's baseball dataset
import pandas as pd
import numpy as np

df = pd.read_csv('https://raw.githubusercontent.com/dk28yip/MSDS7331_lab1/main/Batting.csv') # read in the csv file
df_2015 = df[(df['yearID'] == 2015)]
df_2015.head()

playerID  yearID  stint teamID lgID   G   AB   R   H  2B  ...   RBI  \
107429  abreual01    2020      1    NYA   AL   2    0   0   0   0  ...   0.0   
107430  abreubr01    2020      1    HOU   AL   4    0   0   0   0  ...   0.0   
107431  abreujo02    2020      1    CHA   AL  60  240  43  76  15  ...  60.0   
107432  acunaro01    2020      1    ATL   NL  46  160  46  40  11  ...  29.0   
107433  adamewi01    2020      1    TBA   AL  54  185  29  48  15  ...  23.0   

         SB   CS  BB    SO  IBB  HBP   SH   SF  GIDP  
107429  0.0  0.0   0   0.0  0.0  0.0  0.0  0.0   0.0  
107430  0.0  0.0   0   0.0  0.0  0.0  0.0  0.0   0.0  
107431  0.0  0.0  18  59.0  1.0  3.0  0.0  1.0  10.0  
107432  8.0  1.0  38  60.0  2.0  4.0  0.0  0.0   3.0  
107433  2.0  1.0  20  74.0  0.0  0.0  0.0  0.0   4.0  

[5 rows x 22 columns]

• Data Understanding (80 points total)
• [10 points] Describe the meaning and type of data (scale, values, etc.) for each
attribute in the data file.
• [15 points] Verify data quality: Explain any missing values, duplicate data, and outliers.
Are those mistakes? How do you deal with these problems? Be specific.
• [10 points] Give simple, appropriate statistics (range, mode, mean, median, variance,
counts, etc.) for the most important attributes and describe what they mean or if you
found something interesting. Note: You can also use data from other sources for
comparison. Explain the significance of the statistics run and why they are meaningful.
• [15 points] Visualize the most important attributes appropriately (at least 5 attributes).
Important: Provide an interpretation for each chart. Explain for each attribute why the
chosen visualization is appropriate.
• [15 points] Explore relationships between attributes: Look at the attributes via scatter
plots, correlation, cross-tabulation, group-wise averages, etc. as appropriate. Explain
any interesting relationships.
• [10 points] Identify and explain interesting relationships between features and the class
you are trying to predict (i.e., relationships with variables and the target classification).
• [5 points] Are there other features that could be added to the data or created from
existing features? Which ones?

[10 points] Describe the meaning and type of data (scale, values, etc.) for each
attribute in the data file.

In [15]:
#following code will describe the data
df_2020.describe()

yearID        stint            G           AB            R  \
count  1360.0  1360.000000  1360.000000  1360.000000  1360.000000   
mean   2020.0     1.053676    19.647794    43.404412     6.135294   
std       0.0     0.231897    16.825038    67.214644    10.397936   
min    2020.0     1.000000     1.000000     0.000000     0.000000   
25%    2020.0     1.000000     6.000000     0.000000     0.000000   
50%    2020.0     1.000000    14.000000     0.000000     0.000000   
75%    2020.0     1.000000    28.000000    70.500000     9.000000   
max    2020.0     3.000000    60.000000   248.000000    51.000000   

                 H           2B           3B           HR          RBI  \
count  1360.000000  1360.000000  1360.000000  1360.000000  1360.000000   
mean     10.616912     2.075735     0.177206     1.694118     5.866176   
std      17.596144     3.691288     0.554793     3.393362    10.311186   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%      16.000000     3.000000     0.000000     2.000000     8.000000   
max      78.000000    23.000000     6.000000    22.000000    60.000000   

                SB           CS           BB           SO          IBB  \
count  1360.000000  1360.000000  1360.000000  1360.000000  1360.000000   
mean      0.650735     0.214706     4.479412    11.460294     0.148529   
std       1.838013     0.659624     8.076320    18.006414     0.665560   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     6.000000    19.000000     0.000000   
max      24.000000     8.000000    49.000000    90.000000    12.000000   

               HBP           SH           SF         GIDP  
count  1360.000000  1360.000000  1360.000000  1360.000000  
mean      0.603676     0.092647     0.295588     0.909559  
std       1.410405     0.434297     0.747413     1.752508  
min       0.000000     0.000000     0.000000     0.000000  
25%       0.000000     0.000000     0.000000     0.000000  
50%       0.000000     0.000000     0.000000     0.000000  
75%       1.000000     0.000000     0.000000     1.000000  
max      14.000000     5.000000     6.000000    10.000000

In [18]:
print (df_2020.dtypes)
print (df_2020.info())

playerID     object
yearID        int64
stint         int64
teamID       object
lgID         object
G             int64
AB            int64
R             int64
H             int64
2B            int64
3B            int64
HR            int64
RBI         float64
SB          float64
CS          float64
BB            int64
SO          float64
IBB         float64
HBP         float64
SH          float64
SF          float64
GIDP        float64
dtype: object
<class 'pandas.core.frame.DataFrame'>
Int64Index: 1360 entries, 107429 to 108788
Data columns (total 22 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   playerID  1360 non-null   object 
 1   yearID    1360 non-null   int64  
 2   stint     1360 non-null   int64  
 3   teamID    1360 non-null   object 
 4   lgID      1360 non-null   object 
 5   G         1360 non-null   int64  
 6   AB        1360 non-null   int64  
 7   R         1360 non-null   int64  
 8   H         1360 non-null   int64  
 9 

(1360, 22)

In [19]:
display(df_2020.shape)

(1360, 22)

The dataset has 1360 rows and 22 columns

# Exceptional Work (10 points total)

In [ ]:
You have free reign to provide additional analyses.
One idea: implement dimensionality reduction, then visualize and interpret the results.

In [1]:
from sklearn.manifold import TSNE
from sklearn.decomposition import PCA
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis as LDA
from sklearn.preprocessing import StandardScaler
from matplotlib.pyplot import scatter
import plotly
from plotly.graph_objs import Scatter, Marker, Layout, layout,XAxis, YAxis, Bar, Line
%matplotlib inline

In [ ]:
# Use PCA from sklearn.decomposition
pca = PCA(n_components=2)
lda = LDA(n_components=2)
# Fit data and transform it
pca.fit(X)
X_pca = pca.transform(X)
X_lda = lda.fit(X, Y).transform(X)

In [ ]:
print ('pca:', pca.components_)
print ('lda:', lda.scalings_.T)